src/main/scala/uk/gov/nationalarchives/tdr/GraphQLClient.scala

```diff
@@ -23,41 +24,67 @@ class GraphQLClient[Data, Variables](url: String)(implicit val ec: ExecutionCont
 
   implicit val customConfig: Configuration = Configuration.default.withDefaults
 
-  case class GraphqlData(data: Option[Data], errors: List[GraphqlError] = Nil)
+  case class GraphqlData(data: Option[Data], errors: List[Error] = Nil)
 
-  def getResult(token: BearerAccessToken, document: Document, variables: Option[Variables] = Option.empty): Future[GraphqlData] = {
+  def getResult(token: BearerAccessToken, document: Document, variables: Option[Variables] = Option.empty): Future[GraphQlResponse[Data]] = {
     val queryJson: Json = Json.fromString(QueryRenderer.render(document, QueryRenderer.Compact))
     val variablesJson: Option[Json] = variables.map(_.asJson)
     val fields: immutable.Seq[(String, Json)] = List("query" -> queryJson) ++ variablesJson.map("variables" -> _)
 
     val body = Json.obj(fields: _*).printWith(GraphQLClient.jsonPrinter)
 
-    val response: Future[Response[Either[ResponseError[Error], GraphqlData]]] =
+    val response: Future[Response[Either[String, String]]] =
       basicRequest
         .post(uri"$url")
         .auth.bearer(token.getValue)
         .body(body)
         .contentType(MediaType.ApplicationJson)
-        .response(asJson[GraphqlData]).send()
+        .response(asString)
```

Comment:
I've split up the HTTP call and the decoding. It makes it easier to debug a decoding error if we have the original response body string.